src/module-elasticsuite-core/Index/Mapping/Field.php

```diff
@@ -78,7 +78,7 @@ public function __construct($name, $type = self::FIELD_TYPE_KEYWORD, $nestedPath
     {
         $this->name       = (string) $name;
         $this->type       = (string) $type;
-        $this->config     = $fieldConfig + $this->config;
+        $this->config     = array_merge($this->config, $fieldConfig);
```

Comment:
Why replacing the + by an array_merge ?

Reply:
You're right, but I'm a fan of using array functions for array instead of addition operators and I followed phpstorm hints http://joxi.ru/Dr8OK11foqzWZA

Reply:
Yes but + and `array_merge` does not have exactly the same behavior.

Since this piece of code is legacy, and that changing it is not related to the issue you are fixing here, I'd recommend not modifying this line.

This can apply to everything else, I prefer that we don't change legacy code unless it's strictly needed for a bugfix.